include/zephyr/net/buf.h

```diff
@@ -38,14 +38,18 @@ extern "C" {
  *
  *  @param _name Name of the net_buf_simple object.
  *  @param _size Maximum data storage for the buffer.
- */
-#define NET_BUF_SIMPLE_DEFINE(_name, _size)     \
-	uint8_t net_buf_data_##_name[_size];       \
-	struct net_buf_simple _name = {         \
-		.data   = net_buf_data_##_name, \
-		.len    = 0,                    \
-		.size   = _size,                \
-		.__buf  = net_buf_data_##_name, \
+ *  @param ... list of data.
+ */
+#define NET_BUF_SIMPLE_DEFINE(_name, _size, ...)				\
+	BUILD_ASSERT(sizeof((uint8_t []){ __VA_ARGS__ }) <= _size);		\
+	COND_CODE_1(sizeof((uint8_t []){ __VA_ARGS__ }),			\
```

Comment:
Is this really working as expected? According to the `COND_CODE_1` documentation, the first parameter is expected to expand to 1 or not, so it is evaluated during pre-processing. But what you have here is expected to be evaluated at compile time.

Reply:
Yes, you are right, so separate macro defined. Thanks.

Reply:
Use `IS_EMPTY(__VA_ARGS__)`.